google/cloud/bigtable/examples/opencensus/bigtable_opencensus.cc

```diff
@@ -0,0 +1,159 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! [all code]
+
+//! [bigtable includes]
+#include "google/cloud/bigtable/table.h"
+#include "google/cloud/bigtable/table_admin.h"
+//! [bigtable includes]
+
+#include "opencensus/exporters/stats/stackdriver/stackdriver_exporter.h"
+#include "opencensus/exporters/stats/stdout/stdout_exporter.h"
+#include "opencensus/exporters/trace/stackdriver/stackdriver_exporter.h"
+#include "opencensus/exporters/trace/stdout/stdout_exporter.h"
+#include "opencensus/stats/stats.h"
+#include "opencensus/trace/sampler.h"
+#include "opencensus/trace/trace_config.h"
+#include <grpcpp/opencensus.h>
+
+int main(int argc, char* argv[]) try {
+  if (argc != 4) {
+    std::string const cmd = argv[0];
+    auto last_slash = std::string(cmd).find_last_of('/');
+    std::cerr << "Usage: " << cmd.substr(last_slash + 1)
+              << " <project_id> <instance_id> <table_id>" << std::endl;
+    return 1;
+  }
+
+  std::string const project_id = argv[1];
+  std::string const instance_id = argv[2];
+  std::string const table_id = argv[3];
+
+  // Register the OpenCensus gRPC plugin to enable stats and tracing in gRPC.
+  grpc::RegisterOpenCensusPlugin();
+
+  // Trace all outgoing RPCs.
```

Comment:
Please note that `ProbabilitySampler(1.0)` is not something that ought to be used for production, and is used for sample purposes only.  Ideally, there ought to be a link to what `ProbabilitySampler` does.

Reply:
Added more comments and provided couple of reference to opencensus that talks about sampling.